lib/orbit/ocs/stream/pipeline.ex

```diff
@@ -96,4 +119,87 @@ defmodule Orbit.Ocs.Stream.Pipeline do
     # This is required per the framework, but could be useful down the line
     # to track which messages actually make it all the way into state.
   end
+
+  # Kinesis Stream Persistence
+
+  defp get_producer_opts(opts) do
+    resume_position = load_resume_position()
+    opts ++ [state: %{resume_position: resume_position}]
+  end
+
+  @spec load_resume_position() :: BroadwayKinesis.SubscribeToShard.starting_position()
+  defp load_resume_position do
+    # Use midnight of current service date as cutoff for determining if the
+    # prior Kinesis resume position is relevant
+    midnight =
+      DateTime.new!(
+        Util.Time.current_service_date(),
+        ~T[00:00:00],
+        Util.Time.current_timezone()
+      )
+
+    with stream_name when not is_nil(stream_name) <- get_stream_name(),
+         stream_state when not is_nil(stream_state) <- load_stream_state(stream_name),
+         false <- expired?(stream_state, midnight) do
+      Logger.info("ocs_kinesis_start sequence=#{stream_state.resume_position}")
+
+      {:after_sequence_number, stream_state.resume_position}
+    else
+      _ ->
+        Logger.info("ocs_kinesis_start timestamp=#{midnight}")
+        {:at_timestamp, midnight}
+    end
+  end
+
+  @spec load_stream_state(String.t()) :: KinesisStreamState.t() | nil
+  defp load_stream_state(stream_name) do
+    stream_state = Repo.get_by(KinesisStreamState, stream_name: stream_name)
+
+    if stream_state == nil do
+      Logger.warning("ocs_kinesis_start stream_name=#{stream_name} : No prior sequence stored")
+    end
+
+    stream_state
+  end
+
+  @spec expired?(KinesisStreamState.t(), DateTime.t()) :: boolean()
+  defp expired?(%KinesisStreamState{last_message_timestamp: timestamp}, cutoff) do
+    expired = :gt == DateTime.compare(cutoff, timestamp)
+
+    if expired do
+      Logger.warning(
+        "ocs_kinesis_start last_timestamp=#{timestamp} cutoff=#{cutoff} : Last timestamp is older than cutoff"
+      )
+    end
+
+    expired
+  end
+
+  @spec persist_resume_position(String.t(), DateTime.t()) :: :ok
+  defp persist_resume_position(resume_position, last_message_timestamp) do
+    stream_name = get_stream_name()
+
+    if stream_name != nil do
+      {:ok, _} =
```

Comment:
suggestion: this fails fast, but do we want to bubble up a `:error` to `handle_message`?